src/main/java/com/example/bookstore/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.example.bookstore.service.impl;
+
+import com.example.bookstore.dto.category.CategoryCreateRequestDto;
+import com.example.bookstore.dto.category.CategoryDto;
+import com.example.bookstore.exception.EntityNotFoundException;
+import com.example.bookstore.mapper.CategoryMapper;
+import com.example.bookstore.model.Category;
+import com.example.bookstore.repository.category.CategoryRepository;
+import com.example.bookstore.service.BookService;
+import com.example.bookstore.service.CategoryService;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+    private final BookService bookService;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        Category category = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Sorry! There is no category with id {"
+                        + id + "} in the DB!")
+        );
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto save(CategoryCreateRequestDto categoryDto) {
+        Category category = categoryMapper.toEntity(categoryDto);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public CategoryDto update(Long id, CategoryCreateRequestDto categoryDto) {
+        getById(id);
```

Comment:
Why do you need to load entity into the persistence context if you don't use it? Better just `exists` method